var/spack/repos/builtin/packages/amdfftw/package.py

```diff
@@ -30,97 +31,23 @@ class Amdfftw(FftwBase):
-    depends_on('automake', type='build', when='+pfft_patches')
-    depends_on('autoconf', type='build', when='+pfft_patches')
-    depends_on('libtool', type='build', when='+pfft_patches')
+ 
     depends_on('texinfo')
 
-    # https://github.com/FFTW/fftw3/commit/902d0982522cdf6f0acd60f01f59203824e8e6f3
-    conflicts('%gcc@8:8.9999', when="@3.3.7")
-    conflicts('precision=long_double', when='@2.1.5',
-              msg='Long double precision is not supported in FFTW 2')
-    conflicts('precision=quad', when='@2.1.5',
-              msg='Quad precision is not supported in FFTW 2')
-    conflicts('+openmp', when='%apple-clang', msg="Apple's clang does not support OpenMP")
-
-    provides('fftw-api@2', when='@2.1.5')
-    provides('fftw-api@3', when='@3:')
-
-    @property
-    def libs(self):
-
-        # Reduce repetitions of entries
-        query_parameters = list(llnl.util.lang.dedupe(
-            self.spec.last_query.extra_parameters
-        ))
-
-        # List of all the suffixes associated with float precisions
-        precisions = [
-            ('float', 'f'),
-            ('double', ''),
-            ('long_double', 'l'),
-            ('quad', 'q')
-        ]
-
-        # Retrieve the correct suffixes, or use double as a default
-        suffixes = [v for k, v in precisions if k in query_parameters] or ['']
-
-        # Construct the list of libraries that needs to be found
-        libraries = []
-        for sfx in suffixes:
-            if 'mpi' in query_parameters and '+mpi' in self.spec:
-                libraries.append('libfftw3' + sfx + '_mpi')
-
-            if 'openmp' in query_parameters and '+openmp' in self.spec:
-                libraries.append('libfftw3' + sfx + '_omp')
-
-            libraries.append('libfftw3' + sfx)
-
-        return find_libraries(libraries, root=self.prefix, recursive=True)
-
-    def patch(self):
-        # If fftw/config.h exists in the source tree, it will take precedence
-        # over the copy in build dir.  As only the latter has proper config
-        # for our build, this is a problem.  See e.g. issue #7372 on github
-        if os.path.isfile('fftw/config.h'):
-            os.rename('fftw/config.h', 'fftw/config.h.SPACK_RENAMED')
-
-    def autoreconf(self, spec, prefix):
-        if '+pfft_patches' in spec:
-            autoreconf = which('autoreconf')
-            autoreconf('-ifv')
-
-    @property
-    def selected_precisions(self):
-        """Precisions that have been selected in this build"""
-        return self.spec.variants['precision'].value
-
     def configure(self, spec, prefix):
         # Base options
         options = [
             '--prefix={0}'.format(prefix),
             '--enable-shared',
             '--enable-threads',
 	    '--enable-sse2',
-            '--enable-avx',
-            '--enable-avx2',
-            '--enable-mpi',
-            '--enable-openmp',
-            '--enable-amd-opt'
+	    '--enable-avx',
+	    '--enable-avx2',
```

Comment:
These flags all depend on the hardware. Are these flags even needed now that Spack has specific microarch support and automatically injects the appropriate flags via the compiler wrappers?